exporters/common/src/main/java/io/opentelemetry/exporter/internal/marshal/MarshalerWithSize.java

```diff
@@ -18,6 +18,11 @@ protected MarshalerWithSize(int size) {
     this.size = size;
   }
 
+  /** Vendored {@link Byte#toUnsignedInt(byte)} to support Android. */
```

Comment:
Based on [your comment here](https://github.com/open-telemetry/opentelemetry-java/pull/6167#discussion_r1460462626) you pointed out the sign problem in the initial PR submission. Great catch too! In order to use the same one-liner in the other places, I could have either just duplicated this implementation to exist in 3 places (`SpanMarshaller`, `SpanLinkMarshaller`, `LogMarshaller`), or instead pull it up to the parent of all 3 for reuse. I chose the latter. 

The comment is 3 years old, I assume it's still true tho (that `Byte.toUnsignedInt()` may not exist on versions of Android that we support). I haven't looked.

Reply:
Oh I see. You just moved to a central place and carried the existing comment with it. [Serializer](https://github.com/open-telemetry/opentelemetry-java) is a more appropriate place, since it contains all sorts of related methods.

Reply:
Ok, I removed it from `MarshallerWithSize`. Instead, added new method called `serializeByteAsFixed32` on `Serializer` and similarly added `MarshalerUtil.sizeByteAsFixed32`. This duplicates the bitmask operation, but I think it's fine.